app/code/Magento/Catalog/Model/Product/Gallery/UpdateHandler.php

```diff
@@ -33,8 +33,10 @@ protected function processDeletedImages($product, array &$images)
             if (!empty($image['removed'])) {
                 if (!empty($image['value_id']) && !isset($picturesInOtherStores[$image['file']])) {
                     $recordsToDelete[] = $image['value_id'];
-                    // only delete physical files if they are not used by any other products
-                    if (!$this->resourceModel->countImageUses($image['file']) > 1) {
+                    $catalogPath = $this->mediaConfig->getBaseMediaPath();
+                    $isFile = $this->mediaDirectory->isFile($catalogPath . $image['file']);
+                    // only delete physical files if they are not used by any other products and if this file exist
+                    if (!($this->resourceModel->countImageUses($image['file']) > 1) && $isFile) {
```

Comment:
Check "$isFile" is more faster, so would be great to swap them.

Reply:
Can you explain me again?. I don't understand well.
Do you want say me that I must not use the variable $isFile and I must specify it in the same line?. 

```// only delete physical files if they are not used by any other products and if this file exist
   if (!($this->resourceModel->countImageUses($image['file']) > 1) && $this->mediaDirectory->isFile($this->mediaConfig->getBaseMediaPath() . $image['file']);) {
```

Reply:
@raumatbel code `if ($isFile && !($this->resourceModel->countImageUses($image['file']) > 1))` would be more optimal from performance perspective as if this is not a file then usage of image will not be checked.
I left up to you to make this fix or not as it may be considered as premature optimization. I think in code like this readability has higher priority.